tools/Deployment/lib/common.ts

```diff
@@ -0,0 +1,157 @@
+
+        let cwd = process.cwd();
+        if (workDir) {
+            if (!path.isAbsolute(workDir)) {
+                workDir = path.join(cwd, workDir);
+            }
+            if (!fs.existsSync(workDir)) {
+                throw new Error(`Can't find ${workDir}.`);
+            }
+
+            process.chdir(workDir);
+        }
+
+        let promise = cp.spawn(command, args);
+        let childProcess = promise.childProcess;
+        childProcess.stdout.on("data", (data) => {
+            process.stdout.write(data.toString());
+        });
+        childProcess.stderr.on("data", (data) => {
+            process.stderr.write(data.toString());
+        })
+        return promise.then(() => {
+            process.chdir(cwd);
+        });
+    }
+
+    static zipAssests(assetFolder: string, targetPath: string) {
+        Guard.argumentNotNullOrEmpty(assetFolder, "assetFolder");
+        Guard.argumentNotNullOrEmpty(targetPath, "targetPath");
+
+        let zip = new jszip();
+
+        fs.readdirSync(assetFolder).forEach(file => {
+            let filePath = path.join(assetFolder, file);
+            if (fs.lstatSync(filePath).isFile()) {
+                let ext = path.extname(filePath);
+                if (ext !== '.xml' && ext !== '.pdb') {
+                    let content = fs.readFileSync(filePath);
+                    zip.file(file, content);
+                }
+            }
+        });
+
+        let buffer = zip.generate({ type: "nodebuffer", compression: "DEFLATE" });
+
+        if (fs.existsSync(targetPath)) {
+            fs.unlinkSync(targetPath);
+        }
+
+        fs.writeFileSync(targetPath, buffer);
+    }
+
+    static computeSha1FromZip(zipPath: string): string {
+        Guard.argumentNotNullOrEmpty(zipPath, "zipPath");
+
+        if (!fs.existsSync(zipPath)) {
+            throw new Error(`${zipPath} doesn't exist.`);
+        }
+
+        let buffer = fs.readFileSync(zipPath);
+        return sha1(buffer);
+    }
+
+    static getVersionFromReleaseNote(releaseNotePath: string): string {
+        Guard.argumentNotNullOrEmpty(releaseNotePath, "releaseNotePath");
+
+        if (!fs.existsSync(releaseNotePath)) {
+            throw new Error(`${releaseNotePath} doesn't exist.`);
+        }
+
+        let regex = /\(Current\s+Version:\s+[vV]([\d\.]+)\)/;
+        let content = fs.readFileSync(releaseNotePath, "utf8");
+
+        let match = regex.exec(content);
+        if (!match || match.length < 2) {
+            throw new Error(`Can't parse version from ${releaseNotePath} in current version part.`);
+        }
+
+        return match[1].trim();
+    }
+
+    static getDescriptionFromReleaseNote(releaseNotePath: string): string {
+        Guard.argumentNotNullOrEmpty(releaseNotePath, "releaseNotePath");
+
+        if (!fs.existsSync(releaseNotePath)) {
+            throw new Error(`${releaseNotePath} doesn't exist.`);
+        }
+
+        let regex = /---{3,}\r?\n([\s\S]+?)[vV][\d\.]+\r?\n---{3,}\r?\n/;
```

Comment:
>regex [](start = 12, length = 5)

the regex looks to match sth. similar to
```
-----

Hello v123
-----

```
why is the description wrapped by 5+ `-` and must append version number?